src/client/views/no-data-view/no-data-view.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2015-2016 Imply Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require('./no-data-view.css');
+
+import * as React from 'react';
+import { User, Customization, AppSettings } from '../../../common/models/index';
+import { STRINGS } from '../../config/constants';
+import { Fn } from '../../../common/utils/general/general';
+
+import { AddCollectionModal } from '../../modals/index';
+import { SvgIcon } from '../../components/index';
+
+import { NoDataHeaderBar} from './no-data-header-bar/no-data-header-bar';
+
+export interface NoDataViewProps extends React.Props<any> {
+  user?: User;
+  appSettings?: AppSettings;
+  onNavClick?: Fn;
+  onOpenAbout: Fn;
+  customization?: Customization;
+}
+
+export interface NoDataViewState {
+  mode?: 'no-connected-data' | 'no-cube';
+}
+
+export class NoDataView extends React.Component <NoDataViewProps, NoDataViewState> {
+
+  constructor() {
+    super();
+    this.state = {};
+  }
+
+  componentWillReceiveProps(nextProps: NoDataViewProps) {
+    const { clusters } = nextProps.appSettings;
+
+    if (!clusters || clusters.length) {
+      this.setState({
+        mode: 'no-connected-data'
+      });
+    } else {
+      this.setState({
+        mode: 'no-cube'
+      });
+    }
+  }
+
+  goToSettings() {
+    window.location.hash = '#settings';
+  }
+
+  renderSettingsIcon() {
+    const { user } = this.props;
+    if (!user || !user.allow['settings']) return null;
+
+    return <div className="icon-button" onClick={this.goToSettings.bind(this)}>
+      <SvgIcon svg={require('../../icons/full-settings.svg')}/>
+    </div>;
+  }
+
+  renderTitle(mode: 'no-connected-data' | 'no-cube'): JSX.Element {
```

Comment:
Dunno, I like functions that don't depend on the context... But in this case I didn't really put that many thoughts in it ^^